### DISCUSSION TITLE
How to use Basic/Digest Authentication with http or socket server

### DISCUSSION THREAD

**Discussion by imohsen7** [reactphp/reactphp#487] [Q&A]
Hi
is there any way to integrate a middleware to use basic or digest authentication for http or socket server with reactPHP ? 
i had make a listener for requests of devices but right now need to protect this listener with authentication . because reactphp run as console script there is no way to use common ways to use basic or digest auths in php . 

is there anyone who drag me out of this ?

**Comment by SimonFrings** [comment]
Hey @imohsen7, thanks for bringing this up :+1: 

You can take a look at the [PSR-15 middlware adapter for reactphp/http](https://github.com/friends-of-reactphp/http-middleware-psr15-adapter), this could help you progressing. You can also go the manual way and use the [`Authorization` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) for this.

There is currently no in-house implementation for this, but we're planning to support more PSRs in the future (also PSR-15) as we wrote in https://github.com/orgs/reactphp/discussions/472#discussioncomment-3861005.

This is also something we want to achieve in [Framework X](https://github.com/clue/framework-x), so I think it's also worth to take a look there. The [Authorization documentation](https://framework-x.org/docs/integrations/authentication/) is still under constructions so there will be more information on this in the future.

I hope this helps!
